examples/fungible-token/src/lib.rs

```diff
@@ -0,0 +1,341 @@
+/**
+* Fungible Token implementation with JSON serialization.
+* NOTES:
+*  - The maximum balance value is limited by U128 (2**128 - 1).
+*  - JSON calls should pass U128 as a base-10 string. E.g. "100".
+*  - The contract optimizes the inner trie structure by hashing account IDs. It will prevent some
+*    abuse of deep tries. Shouldn't be an issue, once NEAR clients implement full hashing of keys.
+*  - This contract doesn't optimize the amount of storage, since any account can create unlimited
+*    amount of allowances to other accounts. It's unclear how to address this issue unless, this
+*    contract limits the total number of different allowances possible at the same time.
+*    And even if it limits the total number, it's still possible to transfer small amounts to
+*    multiple accounts.
+*/
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_bindgen::collections::Map;
+use near_bindgen::{env, metadata, near_bindgen, AccountId, Balance};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+/// Contains balance and allowances information for one account.
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct Account {
+    /// Current account balance.
+    pub balance: Balance,
+    /// Escrow Account ID hash to the allowance amount.
+    /// Allowance is the amount of tokens the Escrow Account ID can spent on behalf of the account
+    /// owner.
+    pub allowances: Map<Vec<u8>, Balance>,
+}
+
+impl Account {
+    /// Initializes a new Account with 0 balance and no allowances for a given `account_hash`.
+    pub fn new(account_hash: Vec<u8>) -> Self {
+        Self { balance: 0, allowances: Map::new(account_hash) }
+    }
+
+    /// Sets allowance for account `escrow_account_id` to `allowance`.
+    pub fn set_allowance(&mut self, escrow_account_id: &AccountId, allowance: Balance) {
+        let escrow_hash = env::sha256(escrow_account_id.as_bytes());
+        if allowance > 0 {
+            self.allowances.insert(&escrow_hash, &allowance);
+        } else {
+            self.allowances.remove(&escrow_hash);
+        }
+    }
+
+    /// Returns the allowance of account `escrow_account_id`.
+    pub fn get_allowance(&self, escrow_account_id: &AccountId) -> Balance {
+        let escrow_hash = env::sha256(escrow_account_id.as_bytes());
+        self.allowances.get(&escrow_hash).unwrap_or(0)
+    }
+}
+
+//
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct FungibleToken {
+    /// sha256(AccountID) -> Account details.
+    pub accounts: Map<Vec<u8>, Account>,
+
+    /// Total supply of the all token.
+    pub total_supply: Balance,
+}
+
+pub struct U128(u128);
+
+impl From<u128> for U128 {
+    fn from(v: u128) -> Self {
+        Self(v)
+    }
+}
+
+impl Into<u128> for U128 {
```

Comment:
`Into` is usually not implemented directly. Instead we need to implement `From<U128>`